packages/embarkjs/src/blockchain.js

```diff
@@ -367,6 +375,21 @@ Contract.prototype.send = function(value, unit, _options) {
   return this.blockchainConnector.send(options);
 };
 
+Contract.prototype.deployed = function() {
+  if (this.address === undefined && this._deployPromise === undefined) {
+    throw new Error("contract deployment hasn't happened yet");
```

Comment:
Might have to introduce a new error type here so people can check against it.

Reply:
This is a tricky one.

This API sometimes returns a promise, other times it can return an error (async vs sync). I think it's better to only go one way, callers know they can always expect a promise (and then provide the rejection callback).

Reply:
Wow, great catch. Converted it to a promise.